src/app/modules/submission/encrypt-submission/encrypt-submission.service.ts

```diff
@@ -578,3 +580,70 @@ export const getQuarantinePresignedPostData = (
     }),
   )
 }
+
+/**
+ * Invokes lambda to scan the file in the quarantine bucket for viruses.
+ * @param quarantineFileKey object key of the file in the quarantine bucket
+ * @returns okAsync(returnPayload) if file has been successfully scanned with status 200 OK
+ * @returns errAsync(returnPayload) if lambda invocation failed or file cannot be found
+ */
+export const triggerVirusScanning = (quarantineFileKey: string) => {
+  const logMeta = {
+    action: 'triggerVirusScanning',
+    meta: {
+      quarantineFileKey,
+    },
+  }
+  return ResultAsync.fromPromise(
+    AwsConfig.virusScannerLambda.invoke({
+      FunctionName: AwsConfig.virusScannerLambdaFunctionName,
+      Payload: JSON.stringify({ key: quarantineFileKey }),
+    }),
+    (error) => {
+      logger.error({
+        message: 'Error encountered when invoking virus scanning lambda',
+        meta: logMeta,
+        error,
+      })
+
+      return new VirusScanFailedError()
+    },
+  ).andThen((data) => {
+    const { statusCode, body } = JSON.parse(
+      Buffer.from(data?.Payload ?? '').toString(),
+    ) as {
+      statusCode: number
+      body: string
+    }
```

Comment:
suggestion: can try a typeguard here for safety, instead of type assertion

Reply:
Thanks for the suggestion! Type guards added here: [de324a3](https://github.com/opengovsg/FormSG/pull/6734/commits/de324a349964247caf91bdffa20f921d20fcb86d)